src/Effects/EffectMgr.cpp

```diff
@@ -309,6 +310,7 @@ void EffectMgr::changeeffectrt(int _nefx, bool avoidSmash)
     preset = 0;
     memset(efxoutl, 0, synth.bufferbytes);
     memset(efxoutr, 0, synth.bufferbytes);
+    if(sync) sync->detach(efx);
```

Comment:
sync->detach() is called within changeeffectrt, thus it must be realtime safe. std::vector insertions/deletions are not realtime safe. I'd recommend using a fixed length array within Sync. std::algorithms should still be safe.

Reply:
ok.